### ISSUE TITLE
Parsing of meta file parameters for MSFragger not finished?

### ISSUE THREAD

**Issue by scaramonche** [Proteobench/ProteoBench#159]
Not sure if something went wrong with merging or if it was still work in progress, but there is no extract_params method in the fragger.py file.

**Comment by enryH** [comment]
Yes, for now there is only the parsing on the text parameters. I can add the information we currently can extract, but it's not too much. @mlocardpaulet will try to find see if MSFragger has another parameter file we could use (#58)

**Comment by mlocardpaulet** [comment]
> Yes, for now there is only the parsing on the text parameters. I can add the information we currently can extract, but it's not too much. @mlocardpaulet will try to find see if MSFragger has another parameter file we could use (#58)

@brvpuyve is working on an email to MSFragger's dev. team to identify where we can find the parameters. I will check with him how this is going. You can wait for us to get the information if you'd like. Don't waste time looking for these things when we can ask. We can start without FragPipe support.
By the way, Fragpipe would be the software tool and MSFragger the search engine. @enryH could you make sure that we don't mix these names up?

**Comment by brvpuyve** [comment]
Hi all, 

Sorry for the delay! 
First of all, Marie you are correct (yet again). MSFragger is the search engine and FragPipe is the overarching package from the Nesvizhskii lab.

From Fengchao, I received following answer: in the pepXML result file, version number and parameters are listed. I checked and indeed this is the case. @enryH if you want we can go over it together? Or if you need a template file, please let me know.

**Comment by enryH** [comment]
I don't have that much time the coming to weeks to look into the topic. Could you just add an example of the `pepXML` file into the cloud?

**Comment by brvpuyve** [comment]
@enryH Done!!

**Comment by enryH** [comment]
So Bard and I had a look. I'll try to add the missing informations from the `fragpipe.workflow` text file. This will mean that the UI will also need to handle the case for more than one parameter file (if not all information in `fragger.params` is contained in `fragpipe.workflow`)

**Comment by RobbinBouwmeester** [comment]
Would it be an idea to ask the end-user to zip the files before upload?

**Comment by brvpuyve** [comment]
I think it would be nice if we could enable zip upload, but will we have enough storage capacity for this?

**Comment by enryH** [comment]
I did add the fragpipe.workflow parameter file to add more arguments to the extracted parameters, see [here](https://github.com/Proteobench/ProteoBench/pull/198/files#diff-f11842d021a34fc94c3c1cbf5537e9a3bdea2763f690e7433ddeb12b5fa8c77f)

Questions:

- where would i find the MBR information?
- the fixed and variable modifications are written quite differently to other tools. Is there other information?

**Comment by brvpuyve** [comment]
Hi Henry,

- The MBR information, see line 61: ionquant.mbr,1, (1 means MBR is activated)
- Modifications: Fixed => line 160: 57.02146,C (cysteine),true,-1. All the others are by default turned on but the mass shift is set at 0.0 so it will not detect these/ Variable => Line 161: the ones that are active are set to true, in this case being oxidation and N-term Acetyl: "15.9949,M,true,3; 42.0106,[^,true,1;

**Comment by enryH** [comment]
I'll add the MBR information, but I guess for now I won't parse the parameters further. But I'll document the format a bit.

**Comment by mlocardpaulet** [comment]
So now we "just" need to update the documentation to have the information on what file to submit for retrieving the paramters etc... Maybe @brvpuyve can do it?

**Comment by mlocardpaulet** [comment]
I still need to write the documentation: what file to upload how.
